mcux/middleware/mcux-sdk-middleware-usb/device/usb_device_lpcip3511.c

```diff
@@ -8,6 +8,11 @@
 
 #include "usb_device_config.h"
 #include "usb.h"
+/* CONFIG_UDC_DRIVER is for Zephyr, it will not be defined in NXP MCUXpresso SDK */
+#if ((defined CONFIG_UDC_DRIVER) && (CONFIG_UDC_DRIVER))
+#include "fsl_device_registers.h"
+#include "usb_device_mcux_drv_port.h"
+#else
```

Comment:
Why is this change needed. We had no local changes to the SDK driver related to Zephyr earlier. All Zephyr related changes were kept inside the Zephyr driver.

Reply:
Hi Mahesh, I add more explanation in the README for it. The issue is: In order to use the SDK controller driver, Zephyr need to include many SDK device stack header files. I add one `usb_device_mcux_drv_port.h `to reduce the complexity/dependency to only need to include this header file. It makes `udc_mcux.c` to use SDK usb controller drivers more easily and more convenient/maintainable to adapter with Zephyr usb controller driver interface because many SDK structure/variables are not used in Zephyr environment.

Anyway, as I replied to David, I will put this change to SDK USB stack.